includes/internal/services/class-progress-storage-settings.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * File containing the Progress_Storage_Settings class.
+ *
+ * @package sensei
+ */
+
+namespace Sensei\Internal\Services;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Progress_Storage_Settings.
+ *
+ * @internal
+ *
+ * @since $$next-version$$
+ */
+class Progress_Storage_Settings {
+	/**
+	 * Comment-based storage.
+	 *
+	 * @var string
+	 */
+	public const COMMENTS_STORAGE = 'comments';
+
+	/**
+	 * Table-based storage.
+	 *
+	 * @var string
+	 */
+	public const TABLES_STORAGE = 'custom_tables';
+
+	/**
+	 * Get the storage repositories.
+	 *
+	 * @return array Returns an array of repositories where the key is the repository slug and the value is the description.
+	 */
+	public static function get_storage_repositories(): array {
+		return array(
+			self::COMMENTS_STORAGE => __( 'WordPress comments based storage', 'sensei-lms' ),
+			self::TABLES_STORAGE   => __( 'High-Performance progress storage (experimental)', 'sensei-lms' ),
+		);
+	}
+
+	/**
+	 * Returns true if the HPPS feature is enabled.
+	 *
+	 * @return bool
+	 */
+	public static function is_hpps_enabled(): bool {
+		return Sensei()->settings->settings['experimental_progress_storage'] ?? false;
+	}
+
+	/**
+	 * Returns current storage repository.
+	 *
+	 * @return string
+	 */
+	public static function get_current_repository(): string {
+		return Sensei()->settings->settings['experimental_progress_storage_repository'] ?? self::COMMENTS_STORAGE;
+	}
+
+	/**
+	 * Returns true if the comments repository is enabled.
+	 *
+	 * @return bool
+	 */
+	public static function is_comments_repository(): bool {
+		return self::COMMENTS_STORAGE === self::get_current_repository();
+	}
+
+	/**
+	 * Returns true if the tables repository is enabled.
+	 *
+	 * @return bool
+	 */
+	public static function is_tables_repository(): bool {
+		return self::TABLES_STORAGE === self::get_current_repository();
+	}
```

Comment:
Will these methods still return true if the feature is disabled?

Reply:
When the feature is disabled this method (`is_tables_repository`) should return false.
But `is_comments_repository` should return true.

Reply:
For me, this is not the case. I think it still returns true because the settings are not cleared. This should be fixed by https://github.com/Automattic/sensei/pull/7353 so I think it's ok.